src/de.rs

```diff
@@ -117,965 +70,516 @@ where
-                self.parse_map(len as usize, visitor)
-            }
-            0xba => {
-                let len = self.parse_u32()?;
-                self.parse_map(len as usize, visitor)
-            }
-            0xbb => {
-                let len = self.parse_u64()?;
-                if len > usize::max_value() as u64 {
-                    return Err(self.error(ErrorCode::LengthOutOfRange));
-                }
-                self.parse_map(len as usize, visitor)
-            }
-            0xbc..=0xbe => Err(self.error(ErrorCode::UnassignedCode)),
-            0xbf => self.parse_indefinite_map(visitor),
-
-            // Major type 6: optional semantic tagging of other major types
-            // Only tag 42 is supported, hence we refuse parsing any other tags here.
-            0xc0..=0xd7 => Err(self.error(ErrorCode::UnexpectedCode)),
-            0xd8 => {
-                if self.parse_u8()? == CBOR_TAGS_CID {
-                    self.parse_cid(visitor)
-                } else {
-                    Err(self.error(ErrorCode::UnexpectedCode))
-                }
-            }
-            0xd9..=0xdb => Err(self.error(ErrorCode::UnexpectedCode)),
-            0xdc..=0xdf => Err(self.error(ErrorCode::UnassignedCode)),
-
-            // Major type 7: floating-point numbers and other simple data types that need no content
-            0xe0..=0xf3 => Err(self.error(ErrorCode::UnassignedCode)),
-            0xf4 => visitor.visit_bool(false),
-            0xf5 => visitor.visit_bool(true),
-            0xf6 => visitor.visit_none(),
-            // DAG-CBOR doesn't support `undefined`
-            0xf7 => Err(self.error(ErrorCode::UnexpectedCode)),
-            0xf8 => Err(self.error(ErrorCode::UnassignedCode)),
-            0xf9 => {
-                let value = self.parse_f16()?;
-                visitor.visit_f32(value)
-            }
-            0xfa => {
-                let value = self.parse_f32()?;
-                visitor.visit_f32(value)
-            }
-            0xfb => {
-                let value = self.parse_f64()?;
-                visitor.visit_f64(value)
-            }
-            0xfc..=0xfe => Err(self.error(ErrorCode::UnassignedCode)),
-            0xff => Err(self.error(ErrorCode::UnexpectedCode)),
-
-            _ => unreachable!(),
-        }
+        let mut de = self.try_step()?;
+        let map = Accessor::map(&mut de)?;
+        visitor.visit_map(map)
     }
-}
-
-impl<'de, 'a, R> de::Deserializer<'de> for &'a mut Deserializer<R>
-where
-    R: Read<'de>,
-{
-    type Error = Error;
 
     #[inline]
-    fn deserialize_any<V>(self, visitor: V) -> Result<V::Value>
+    fn deserialize_struct<V>(
+        self,
+        _name: &'static str,
+        _fields: &'static [&'static str],
+        visitor: V,
+    ) -> Result<V::Value, Self::Error>
     where
-        V: de::Visitor<'de>,
+        V: Visitor<'de>,
     {
-        self.parse_value(visitor)
+        self.deserialize_map(visitor)
```

Comment:
why does `struct` map to `map`?

Reply:
Because structs are serialized as CBOR maps.